### ISSUE TITLE
aria-checked shouldn't be a mandatory attribute for element with role=switch if the element is created with checkbox. 

### ISSUE THREAD

**Issue by mpnkhan** [dequelabs/axe-core#2225]
There are multiple ways to create a switch component, div, span, button. We can also use native element like checkbox with role=switch. Here is an example with switch component with checkboxes http://hcm-sample.surge.sh. 
aria-checked shouldn't be an required attribute for <input type="checkbox role="switch"">
But Axe validate it as violation. 

Expectation: << Describe what you expect the product to do >>

Actual: << Describe what the product actually does >>

Motivation: << Describe why you want the behavior to be changed >>

<pre><code>
axe-core version: X.Y.Z
axe-webdriver, extension or other integration version: X.Y.Z

Browser and Assistive Technology versions

For Tooling issues:
- Node version: XX  
- Platform:  
</code></pre>

**Comment by straker** [comment]
Thanks for the issue. We'll have to build an exception into the check for `checkbox[role=switch]`

**Comment by padmavemulapati** [comment]
Verified the issue, if we keep aria-cheked - true and i am nt seeing any failure But if this is the case atia-checked is not a mandatory I am still seeing the issue. @straker , is this fix pushed ?

**Comment by padmavemulapati** [comment]
No failure seeing for aria-attributes even not passing aria-checked for role=switch.
![image](https://user-images.githubusercontent.com/47409579/82551370-ae8e4c80-9b7d-11ea-8ca6-366383412022.png)

**Comment by jeankaplansky** [comment]
behind the scenes, no new rule help required.
